src/classes/ALLO_PaymentSync_SVC.cls

```diff
@@ -410,7 +410,7 @@ public class ALLO_PaymentSync_SVC {
             Decimal percentGAU = (totalAllPayments != 0) ? totalPaymentAllocations / totalAllPayments : 0;
 
             // Calculate the proposed amount (based on the Opportunity amount, multiplied by the GAU Percentage)
-            Decimal newGAUAmtForOpp = context.opportunity.Amount * percentGAU;
+            Decimal newGAUAmtForOpp = (context.opportunity.Amount * percentGAU).setScale(18);
```

Comment:
Can you change this to setScale(2)?  This would actually reflect the # of decimal places we actually use, and the actual precision of our data, since we do not do fractional pennies.  I changed it in my local org, and reran the tests.  Nothing broke as a result.  I believe we set the scale to 2 elsewhere in a few other spots, as well.